atc/db/container_repository.go

```diff
@@ -346,19 +346,32 @@ func scanContainer(row sq.RowScanner, conn Conn) (CreatingContainer, CreatedCont
 }
 
 func (repository *containerRepository) DestroyFailedContainers() (int, error) {
-	result, err := sq.Delete("containers").
+	rows, err := selectContainers().
```

Comment:
Might just be something to think about but I'm a little unsure about the fact that we select out all these failed containers and then update the state one by one without using the return value of the `Destroy()` method. It seems like in hindsight it would be easiest and most efficient to just have this `DestroyFailedContainers` method to just straight up update all the failed containers to `destroying` state? Maybe there a reason why we need to do it this way that I'm missing?

Reply:
we saw that all the state transitions were happening through db/container function calls. Its a good point that we can bypass everything; we were just trying to be clean but I guess no one will actually call failedContainer.Destroy() other than container_repo